analyzers/src/SonarAnalyzer.CSharp/SymbolicExecution/Roslyn/SecureRandomSeedsShouldNotBePredictable.cs

```diff
@@ -0,0 +1,163 @@
+        }
+        else if (operation.AsArrayElementReference() is { } arrayElementReference)
+        {
+            return ProcessArrayElementReference(state, arrayElementReference);
+        }
+        else if (operation.AsInvocation() is { } invocation)
+        {
+            return ProcessArraySetValue(state, invocation)
+                ?? ProcessStringToBytes(state, invocation)
+                ?? ProcessSecureRandomGetInstance(state, invocation)
+                ?? ProcessSecureRandomSetSeed(state, invocation)
+                ?? ProcessSecureRandomNext(state, invocation)
+                ?? state;
+        }
+        return state;
+    }
+
+    private static ProgramState ProcessArrayCreation(ProgramState state, IArrayCreationOperationWrapper arrayCreation)
+    {
+        if (arrayCreation.Type.IsAny(KnownType.System_Byte_Array, KnownType.System_Char_Array))
+        {
+            var isConstant = arrayCreation.Initializer.WrappedOperation is null || arrayCreation.Initializer.ElementValues.All(x => x.ConstantValue.HasValue);
+            return state.SetOperationConstraint(
+                arrayCreation,
+                isConstant ? CryptographicSeedConstraint.Predictable : CryptographicSeedConstraint.Unpredictable);
+        }
+        return state;
+    }
+
+    private static ProgramState ProcessArrayElementReference(ProgramState state, IArrayElementReferenceOperationWrapper arrayElementReference) =>
+        arrayElementReference.IsAssignmentTarget() || arrayElementReference.IsCompoundAssignmentTarget()
+            ? state.SetSymbolConstraint(arrayElementReference.ArrayReference.TrackedSymbol(state), CryptographicSeedConstraint.Unpredictable)
+            : state;
+
+    private static ProgramState ProcessArraySetValue(ProgramState state, IInvocationOperationWrapper invocation)
+    {
+        if (invocation.TargetMethod.Name == nameof(Array.SetValue)
+            && invocation.TargetMethod.ContainingType.Is(KnownType.System_Array))
+        {
+            return invocation.ArgumentValue("value") is { ConstantValue.HasValue: true }
+                ? state
+                : state.SetSymbolConstraint(invocation.Instance.TrackedSymbol(state), CryptographicSeedConstraint.Unpredictable);
+        }
+        return null;
+    }
+
+    private static ProgramState ProcessSecureRandomGetInstance(ProgramState state, IInvocationOperationWrapper invocation) =>
+        invocation.TargetMethod.Name == "GetInstance"
+        && invocation.TargetMethod.ContainingType.DerivesFrom(KnownType.Org_BouncyCastle_Security_SecureRandom)
+        && invocation.Arguments.Length == 2
+        && invocation.ArgumentValue("autoSeed") is { ConstantValue: { HasValue: true, Value: false } }
+            ? state.SetOperationConstraint(invocation, CryptographicSeedConstraint.Predictable)
+            : null;
+
+    private ProgramState ProcessStringToBytes(ProgramState state, IInvocationOperationWrapper invocation)
```

Comment:
This processes 4 cases:

```csharp
Encoding.UTF42.GetBytes(string s);
Encoding.UTF42.GetBytes(char[] chars);

Convert.FromBase64String(string s);
Convert.FromBase64CharArray(char[] inArray, int, int);
```

I know that this method looks like a lot, but I think it is more readable to have all of them in the same place, as they are parsed very similarily.